### ISSUE TITLE
Do not send "Accept Chat" message to all operators

### ISSUE THREAD

**Issue by Vintic** [LiveHelperChat/telegram#26]
1. Send "Accept Chat" button when operator is assigned to chat. And only to this operator.

2. Also replace all sent "Accept Chat" buttons (related to this chat, different operators) to message like "Chat already accepted".

**Comment by remdex** [comment]
1. This one is not so simple as as it sounds. As operator usually is assigned only after chat is started. Which I do not think you do. So it would required some delay and with delay comes cronjobs/background workers.

2. Does not make sense because if according to first point we should not send it at all... But n general does telegram allows to change already send message content? If so why don't you extend extension itself? :D

**Comment by Vintic** [comment]
I just want, that telegram behavior would be same as LiveHelperChat site.

1. Yes, in widget I set "Operator ID" and "Automatically assign chat to another operator if operator did not accepted chat in seconds" in Department settings

2. With options mentioned in 1. Chat button should appear/disappear for different operators. Yes message can be changed using edit_message_text(chat_id, message_id, text). Oh, I would like, but it would take a long time.

**Comment by remdex** [comment]
For first one i can do that pretty quickly. Will do till friday but for second one for me also would take long time as i would have to store sended message id and so on :)

**Comment by remdex** [comment]
Please try to update.
1. It will send message only if operator is not assigned to chat or operator is the one to whom chat is assigned.
2. also will send message on auto assignment workflow to operator to whom chat was assigned.

**Comment by Vintic** [comment]
Thanks, It works, but: 
1. It send message to telegram only if user is online.
2. Redirect message only if exist another online user. ("Automatically assign chat to another operator if operator did not accepted chat in seconds" option)
3. Even when I set working hours, (chat is online, user offline). It wont send message to telegram, and redirect messages.

How, can I set All users to be online?

**Comment by remdex** [comment]
1. Operator has to be checked that he is online just. If in web back office he checked he's offline he won't get offline. 
2. This one did not understood.
3. It's not related to telegram operators statues.

**Comment by Vintic** [comment]
If I do logout, will user remain online?
Reassign don't work now if department users are logged out.

Story: 
I created department with user and his manager. Set both online. Both connected to telegram. And both logged out. 
Set working hours for dep.. Set reassign after 30sec. Added widget with user assign.
Now message is sent to assigned user, after 30 sec. manager don't receive it. When I login to Manager account, chat will reasign to him and message wil be sent to his telegram.

**Comment by remdex** [comment]
Right I see. But i'm afraid you can't force for operator to be visible always at the moment. And it's not related to telegram but more to LHC itself as we would need another flag always online. Or you can just write small cronjob which would do something like


"UPDATE lh_userdep SET last_activity = <unix_timestamp>";

**Comment by Vintic** [comment]
What the purpose of the telegram plugin if you always should be logged in Web Interface?
Or I'm doing something wrong?

**Comment by remdex** [comment]
Telegram plugin was not foreseen to work with Auto assignment workflow as you are doing that. As for most cases it's fine for majority of users and what you can change to have auto assignment working I already wrote.

**Comment by Vintic** [comment]
OK. Thank you very much! I think for that should be created another issue on LHC itself.

**Comment by Vintic** [comment]
Again all Department operators receive request to accept chat, not only person that was assigned in widget.

**Comment by remdex** [comment]
It's not possible.
https://github.com/LiveHelperChat/telegram/blob/master/bootstrap/bootstrap.php#L175

If you reproduce create a pull request with fix and I'll gladly accept it :)

**Comment by Vintic** [comment]
ok, I will try, now I see that at first message is sent to correct user, then in a second, it is assigned to another available user. And message is different: "Chat was assigned to you".

It's fresh installation. 
In department setting is set only, Automatically assign chat to another operator if operator did not accepted chat in "30" seconds

**Comment by Vintic** [comment]
Same without telegram bot, on fresh install.
1. Create 2 users, 
2. Set same department
3. Automatically assign chat to another operator if operator did not accepted chat in "30" seconds
4. Create widget with assign to operator Nr. 1.
5. Login for both operator
6. Start chat.
Chat will start with operator Nr.1 and in a second with operator Nr. 2

**Comment by remdex** [comment]
That's more usefull description of the issue. I'll try it out your scenario today or tomorrow and let you know :)

**Comment by remdex** [comment]
Here is your change to try it out :)
https://github.com/LiveHelperChat/livehelperchat/commit/f144bc6c002aa7ce253feb351611bba13b85a9de

**Comment by Vintic** [comment]
It works) Thank you!
